### ISSUE TITLE
How many lanes quest - show direction of lanes in static image too (ie. not only with moving cars)

### ISSUE THREAD

**Issue by mnalis** [streetcomplete/StreetComplete#3581]
**Use case**

(This was mentioned as an aside in https://github.com/streetcomplete/StreetComplete/issues/2350#issuecomment-746957870 and got several likes, but as it wasn't the main issue, it got ignored. So I'm opening it as new enhancement idea).

In `How many lanes for cars does this road have?` quest, it is often hard to follow what lanes go in what direction.

While moving cars are nice eye candy, from usability perspective they do not work that well (at least for me), especially if there is 3 or more lanes altogether. So I need to concentrate and wait for several rounds of cars to pass before I'm reasonably sure what is actual lanes configuration that is being shown by SC. Which is both time consuming and error prone (again, at least for me - I guess multiple animation tracking is easier for younger folks).

**Proposed Solution**

May I suggest also marking lanes in pictures **not only** with moving cars, but **also with static arrows**? Something like this: 
![lane2](https://user-images.githubusercontent.com/156656/102401622-eeb69000-3fe3-11eb-8573-9ac8b7087b8b.png)

That should improve readability/accessibility of that quest, with not much side effects (please ignore submenu in the bottom of the screenshot - it was used on example from another proposed idea in that ticket)

In other comments it was suggested to make static arrows bigger, rounder, and/or more transparent / different color (like blue), so they do not get confused with "lane turn" road markings. I'm fine with either, as long as it is **_not only moving objects_** that clearly indicate lane directions.

**Comment by HolgerJeromin** [comment]
I tried to remove confusing with lane turns with transparent arrow heads:
![image](https://user-images.githubusercontent.com/2410353/145562724-a45fc9b2-24c4-4895-beba-bb54c9d28437.png)
Or is this worse?

**Comment by matkoniecz** [comment]
Not entirely sure is it needed...


>  Or is this worse?

It is even more confusing.

Maybe changing color and style of arrow would help? For example a green one.

**Comment by smichel17** [comment]
Hm. In the US, the dividing line is marked in yellow, so this problem does not really exist.

![image](https://user-images.githubusercontent.com/6563664/145588190-77e05b81-139d-4f05-8fe6-800d524e2430.png)

But I see how it is an issue in Croatia

![image](https://user-images.githubusercontent.com/6563664/145588412-72fba34c-1166-4552-b6c1-3bc0944f33df.png)

Maybe we can do something simpler, like making it possible to distinguish between the middle line and the lane dividers. In @mnalis illustration, the lane dividers are dashed. Maybe that would be enough?

**Comment by HolgerJeromin** [comment]
> Maybe we can do something simpler, like making it possible to distinguish between the middle line and the lane dividers. In @mnalis illustration, the lane dividers are dashed. Maybe that would be enough?

That is the case in the stable app right now :)

**Comment by smichel17** [comment]
Ah, I think my debug version must be old :sweat_smile:

**Comment by peternewman** [comment]
It's not done by country is it? I get a solid line between the carriageways and dashed between lanes in the UK, so even if your code is old @smichel17 I'm surprised you're getting two different renders.

**Comment by mnalis** [comment]
With fatter blue arrows it might look like this (quick mockup):

![Screenshot_20211210_193556_de westnordost streetcomplete_s](https://user-images.githubusercontent.com/156656/145626104-f137b1d8-5e23-4afa-9fa0-d04321bb9540.png)

As noted, the main problem for me is (especially when there is a more complicated situation, like more lanes or uneven number of lanes, or other roads closeby, or - god forbid - country in which cars drive on "wrong" side of the road etc.) that I usually need to see several passes of animation in order to make such a mental picture in my head of lanes with arrows, and then correlate that mental picture to reality, which takes time and effort (for me at least). If the arrows were instead drawn by the SC itself (as proposed), that mental load could be greatly reduced.

As to the comments - yes, the quest _**can be solved**_ correctly due to dotted/full lines distinction (especially by more experienced SC mapper) _**even without arrows**_ (the same as it could be solved without moving cars animation) - it is just that **addition of arrows** would (IMHO) made quest **easier** (and thus **less prone to errors**) and **faster** to solve.

**Comment by westnordost** [comment]
Why is it necessary to know the direction in which the cars are going? Why would it be helpful?

**Comment by matkoniecz** [comment]
> Why is it necessary to know the direction in which the cars are going? Why would it be helpful?

Someone who just filled answers and become unsure have they filled it correctly? (that was my motivation for triggering car animation immediately - which is implemented)

**Comment by westnordost** [comment]
But it doesn't matter in which direction the traffic goes, it matters what is on the left and what is on the right side.

In right-hand-traffic countries, the cars will **always** animate from the bottom to the top on the right side and from the top to the bottom on the left side.

**Comment by matkoniecz** [comment]
Someone could be unsure about lane count given for directions.


Or had desire to verify onewayness status.

Personalny, for me animation is sufficient.

**Comment by mnalis** [comment]
The biggest issue is probably with non-symmetric number of lanes (like 2+3, 1+2, etc), where numbers differs. Note that the user might not be oriented in exact direction as the car traffic (so what is on right side of the car passing the road, might not be on the right side of the SC user), and the physical reality of the road is usually not 100% percent aligned with what user sees on the screen (imperfect direction sensors and/or manual rotation etc). Also other roads in vicinity, or RHS user in LHS country (or vice versa) might complicate things even more; and visual feedback helps there.

Thus being sure that one enters correct information is of importance, and indication of lanes direction is helpful there. As @matkoniecz noted, I too always verify what the SC shows matches the reality, before submitting this quest.

Or is your opinion that visual feedback (eg. car animations currently) does not help at all with this quest? I think they do help (even if I personally struggle with animations and am much more at ease with static graphics).

For example, if the quest had no drawing of the road at all but just two numeric input fields named `"Number of lanes on left side of the road"` and `"Number of lanes on right side of the road"` I think it would be much harder (ie. more mentally taxing) to solve for most people - even if such quest format would technically still contain all information needed to answer the quest correctly. 

Thus I think visual feedback is quite useful for this quest - it is just that some people might vastly prefer static image for feedback instead of moving animation, hence this enhancement suggestion.

**Comment by smichel17** [comment]
What if we stop worrying aout how pretty it is and just show numbers the user entered?

![image](https://user-images.githubusercontent.com/6563664/145697587-5bf7123f-6028-44e0-8dc9-5baea83c68bb.png)

**Comment by westnordost** [comment]
@mnalis to be honest, I still don't get it. Yes, visual feedback helps. Of course! This is the idea behind this UI, that you see how the sides entered on the form and on the map relate to each other! And this is exactly the reason why there is no fixed "left side" and no "right side" of the street because it depends on how you rotate the view.

And no matter how you rotate the view, cars will always drive from bottom to top on the right side of the shown illustration and from the top to bottom on the left side of the illustration (on RHS traffic). This is why I say, to have this information specifically pointed out is literally no help at all - what counts is the orientation of the illustration in relation to the map.

So, we are not in the position (yet?!) to discuss what the best UI would be for displaying which side is which, @smichel17 , because we are not even on the same page that this information overall is useful. The cars displayed currently is there for eye-candy, because the app should be fun, not because it is particularly helpful in giving the correct answer.

**Comment by HolgerJeromin** [comment]
Perhaps the animation triggers the _thought_ that it is helpful for double checking the answer. 
So the user wait for it to check if he had done it right. And this is hard for some of us, but we are not able to ignore this additional information.

Without the animation (which I like) the users would depend only on the line markings which are sufficient. This would be faster.

**Comment by riQQ** [comment]
Also, for two lanes it's not possible to determine from the markings if it's one direction with a two lanes or both direction with one lane each. Both have a dashed line in the middle.

**Comment by westnordost** [comment]
It is, because you have to tap once for each side. So you know this is not a oneway-road.

**Comment by riQQ** [comment]
For a road with lanes in both directions I select Marked lanes and then I get a dialog to specify total amount of lanes with a spinner with even values 2, 4, 6, etc..
For a one-way road I select Marked lanes and I get a spinner with values 1, 2, 3 etc..

After selecting 2 and confirming, both forms look the same minus the animation.

**Comment by westnordost** [comment]
Right. You are right.

**Comment by westnordost** [comment]
So, anyway I am not convinced this would be an improvement. Will close this.
